src/multibody/model.hpp

```diff
@@ -124,94 +128,51 @@ namespace se3
+    /// \tparam JointModelDerived The type of the joint model.
     ///
     /// \param[in] parent Index of the parent joint.
-    /// \param[in] j The joint model.
-    /// \param[in] jointPlacement The relative placement of the joint j regarding to the parent joint.
-    /// \param[in] Y Spatial inertia of the body.
-    /// \param[in] effort Maximal joint torque.
-    /// \param[in] velocity Maximal joint velocity.
-    /// \param[in] lowPos Lower joint configuration.
-    /// \param[in] upPos Upper joint configuration.
-    /// \param[in] jointName Name of the joint.
-    /// \param[in] bodyName Name of the body.
-    ///
-    /// \return The index of the new added joint.
+    /// \param[in] joint_model The joint model.
+    /// \param[in] joint_placement Placement of the joint inside its parent joint.
+    /// \param[in] joint_name Name of the joint. If empty, the name is random.
+    /// \param[in] max_effort Maximal joint torque. (Default set to infinity).
+    /// \param[in] max_velocity Maximal joint velocity. (Default set to infinity).
+    /// \param[in] min_config Lower joint configuration. (Default set to infinity).
+    /// \param[in] max_config Upper joint configuration. (Default set to infinity).
     ///
-    template<typename D>
-    JointIndex addJointAndBody(JointIndex parent,const JointModelBase<D> & j,const SE3 & jointPlacement,
-                               const Inertia & Y,
-                               const Eigen::VectorXd & effort, const Eigen::VectorXd & velocity,
-                               const Eigen::VectorXd & lowPos, const Eigen::VectorXd & upPos,
-                               const std::string & jointName = "", const std::string & bodyName = "");
-    
+    /// \return The index of the new joint.
     ///
-    /// \brief Add a Joint with no body (Zero inertia) to the kinematic tree.
-    ///
-    /// \param[in] parent Index of the parent joint.
-    /// \param[in] j The joint model.
-    /// \param[in] jointPlacement The relative placement of the joint j regarding to the parent joint.
-    /// \param[in] jointName Name of the joint.
+    /// \sa Model::appendBodyToJoint
     ///
-    /// \return The index of the new added joint.
-    ///
-    template<typename D>
-    JointIndex addJoint(JointIndex parent,const JointModelBase<D> & j,const SE3 & jointPlacement,
-                        const std::string & jointName = "");
+    template<typename JointModelDerived>
+    JointIndex addJoint(const JointIndex parent, const JointModelBase<JointModelDerived> & joint_model, const SE3 & joint_placement,
+                        const std::string & joint_name = "",
+                        const Eigen::VectorXd & max_effort = Eigen::VectorXd::Constant(JointModelDerived::NV,std::numeric_limits<double>::max()),
+                        const Eigen::VectorXd & max_velocity = Eigen::VectorXd::Constant(JointModelDerived::NV,std::numeric_limits<double>::max()),
```

Comment:
No, there is no restriction. We just need to add a new dedicated visitor for that.